src/finch/autoschedule/optimize.py

```diff
@@ -374,6 +399,58 @@ def propagate_transpose_queries(root):
     return _propagate_transpose_queries(root, bindings={})
 
 
+def normalize_names(root):
+    return _normalize_names(root, counts={})
+
+
+def freshen(counts: dict[str, int], symbol: str) -> str:
```

Comment:
see my comments, could we add the implementation from https://github.com/finch-tensor/finch-tensor-lite/pull/46?

Reply:
We could add it in the same location as #46, in symbolic/environment.py

Reply:
I think what I meant was to copy the file in symbolic/environment.py from that PR to this one, and `from ..symbolic import Namespace` here.

Reply:
Done!